scarlet/component.py

```diff
@@ -525,32 +585,60 @@ def get_model(self, *params):
         model: array
             (Bands, Height, Width) data cube
         """
-        model = np.zeros(self.frame.shape)
-        if len(params):
-            i = 0
-            for k, c in enumerate(self.components):
+
+        # We have to declare the function that inserts sources
+        # into the blend with autograd.
+        # This has to be done each time we fit a blend,
+        # since the number of components => the number of arguments,
+        # which must be linked to the autograd primitive function
+        defvjp(_add_models, *([partial(_grad_add_models, index=k) for k in range(len(self.components))]))
```

Comment:
Still not a fan of this idea because it sets a global number of parameters and prevent correct fitting with multiple concurrent instances of `Blend` in a single thread. Nothing we can do about it now, but I suggest to add a comment here than one can avoid this and revert to the original mode.

Reply:
Also, for clarity, I'd move the entire block of comments and `defvjp` to below the `_add_models` call.

Reply:
Do you mean add this block to just _above_ the `_add_models` call? We have to declare it to `autograd` before we call it.

As for your first comment, I agree that this isn't ideal and I'll add a comment. But the reason I put it here is because we'll only be optimizing one blend at a time, so while the global variable sucks (and jax unfortunately does the same thing) we should never get a conflict, even with multiple blends.